### PR TITLE
kubevirt: create default VM YAML dynamically (create from YAML option)

### PR DESCRIPTION
**Pull request by suomiy**
- do not link to common templates when they get missing

continuation of a discussion in https://github.com/openshift/console/pull/4469 

@yaacov @irosenzw 

cc @mareklibra @jelkosz

### CODE REVIEW

**Comment by suomiy** [comment]
The whole YAML gets generated from the available common templates or default to the hardcoded one when they get missing

aa

File: frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx [hunk 1]
```diff
@@ -43,7 +43,7 @@ import { getVMTemplateNamespacedName } from '../../../selectors/vm-template/sele
 
 const getId = (field: string) => `vm-flavor-modal-${field}`;
 
-const getFlavors = (template: TemplateKind) => {
+const getAvailableFlavors = (template: TemplateKind) => {
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx]: piggybacking this here
@yaacov is this name better?

File: frontend/packages/kubevirt-plugin/src/models/templates/vm-template-yaml.ts [hunk 2]
```diff
@@ -8,14 +8,11 @@ apiVersion: ${TemplateModel.apiGroup}/${TemplateModel.apiVersion}
...
 metadata:
   name: vm-template-example
-  namespace: default
   labels:
     template.kubevirt.io/type: vm
     os.template.kubevirt.io/fedora31: 'true'
-    flavor.template.kubevirt.io/tiny: 'true'
+    flavor.template.kubevirt.io/Custom: 'true'
     workload.template.kubevirt.io/server: 'true'
-    vm.kubevirt.io/template: fedora-server-tiny-v0.7.0
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/models/templates/vm-template-yaml.ts]: removed the pointers to the common templates as this yaml should get used only when the common templates get missing

File: frontend/packages/kubevirt-plugin/src/models/templates/vm-template-yaml.ts [hunk 3]
```diff
@@ -8,14 +8,11 @@ apiVersion: ${TemplateModel.apiGroup}/${TemplateModel.apiVersion}
 kind: ${TemplateModel.kind}
 metadata:
   name: vm-template-example
-  namespace: default
   labels:
     template.kubevirt.io/type: vm
     os.template.kubevirt.io/fedora31: 'true'
-    flavor.template.kubevirt.io/tiny: 'true'
+    flavor.template.kubevirt.io/Custom: 'true'
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/models/templates/vm-template-yaml.ts]: that is why we have to use the `Custom` flavor

File: frontend/packages/kubevirt-plugin/src/models/templates/vm-yaml.ts [hunk 4]
```diff
@@ -8,20 +8,25 @@ apiVersion: ${VirtualMachineModel.apiGroup}/${VirtualMachineModel.apiVersion}
 kind: ${VirtualMachineModel.kind}
 metadata:
   labels:
-    app: example
-    flavor.template.kubevirt.io/tiny: 'true'
+    app: vm-example
+    flavor.template.kubevirt.io/Custom: 'true'
     os.template.kubevirt.io/fedora31: 'true'
-    vm.kubevirt.io/template.revision: '1'
```
Comment (suomiy) [inline frontend/packages/kubevirt-plugin/src/models/templates/vm-yaml.ts]: same here: this file is generated from the code above, but removed the common templates labels and changed flavor to `Custom`

File: frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx (hunk 1 above)
Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx]: :+1: as long as it's different than the global one

**Pull request merged**